app/vmselect/promql/binary_op.go

```diff
@@ -497,7 +503,10 @@ func binaryOpOr(bfa *binaryOpFuncArg) ([]*timeseries, error) {
 			rvs = append(rvs, tssRight...)
 			continue
 		}
-		fillLeftNaNsWithRightValues(tssLeft, tssRight)
+		fillLeftNaNsWithRightValuesOrMerge(tssLeft, tssRight)
+		// tssRight might be filled with NaNs after merge
+		tssRight = dropEmptySeries(tssRight)
```

Comment:
Is it need to make a shallow copy of tssRight? It looks like it can be filtered directly with  removeEmptySeries